src/ring_theory/polynomial/cyclotomic/basic.lean

```diff
@@ -853,4 +860,78 @@ end
 
 end expand
 
+section char_p
+
+/-- If `R` is of characteristic `p` and `¬p ∣ n`, then
+`cyclotomic (n * p) R = (cyclotomic n R) ^ (p - 1)`. -/
+lemma cyclotomic_mul_prime_not_dvd_eq_pow (R : Type*) {p n : ℕ} [hp : fact (nat.prime p)] [ring R]
+  [char_p R p] (hn : ¬p ∣ n) : cyclotomic (n * p) R = (cyclotomic n R) ^ (p - 1) :=
+begin
+  suffices : cyclotomic (n * p) (zmod p) = (cyclotomic n (zmod p)) ^ (p - 1),
+  { rw [← map_cyclotomic _ (algebra_map (zmod p) R), ← map_cyclotomic _ (algebra_map (zmod p) R),
+      this, polynomial.map_pow] },
+  suffices : cyclotomic (n * p) (zmod p) * (cyclotomic n (zmod p)) = (cyclotomic n (zmod p)) ^ p,
+  { have h : cyclotomic n (zmod p) ^ p = cyclotomic n (zmod p) ^ p.pred.succ,
+    { rw [nat.succ_pred_eq_of_pos hp.out.pos] },
+    rw [h, pow_succ, mul_comm (cyclotomic (n * p) _)] at this,
+    exact mul_right_injective₀ (cyclotomic_ne_zero _ _) this },
+  rw [← zmod.expand_card],
+  nth_rewrite 2 [← map_cyclotomic_int],
+  rw [← map_expand, cyclotomic_expand_eq_cyclotomic_mul hp.out hn, polynomial.map_mul,
+    map_cyclotomic, map_cyclotomic]
+end
+
+/-- If `R` is of characteristic `p` and `p ∣ n`, then
+`cyclotomic (n * p) R = (cyclotomic n R) ^ p`. -/
+lemma cyclotomic_mul_prime_dvd_eq_pow (R : Type*) {p n : ℕ} [hp : fact (nat.prime p)] [ring R]
+  [char_p R p] (hn : p ∣ n) : cyclotomic (n * p) R = (cyclotomic n R) ^ p :=
+begin
+  suffices : cyclotomic (n * p) (zmod p) = (cyclotomic n (zmod p)) ^ p,
```

Comment:
i wonder if we can make a lemma/quick tactic for this+int; this is such a common pattern with cyclotomic polys

Reply:
This would be very nice, but I have no idea how to do it.